app/services/subscription_updator.rb

```diff
@@ -0,0 +1,78 @@
+class SubscriptionUpdator
+    class Error < StandardError; end
+    include Ak::Client
+  
+    def self.run(params)
+      new(params).run
+    end
+  
+    def initialize(params)
+      @params = params[:params]
+      @subscription_id = @params[:recurring_id] || raise(ArgumentError)
+      @transaction_id = @params[:trans_id]
+      @amount = @params[:amount]&.to_d || 0
+    end
+  
+    def run
+      handle_subscription_charge()
+    end
+  
+    private
+  
+    def handle_subscription_charge()
+        return unless subscription
+        # Return if a transaction record already exists for this subscription charge. This is to prevent processing
+        # the same subscription charge twice.
+        return unless Payment::Braintree::Transaction.find_by(transaction_id: @transaction_id).blank?
```

Comment:
@osahyoun I have a question about this too.
From what is sent through the webhook listener to the queue, I understand that `transaction_id` is the same as `subscription_id`, correct?
If that is the case, will the `subscription_id` value change for every successful transaction on Braintree? Because if it doesn't change, then the current logic is wrong because we will only save the transaction on DB for the first occurrence, right?

Let me know if I explained myself with this comment; we can discuss it further on a call

Reply:
Dismissing comment, Omar helped me understand how this is currently working and the subscription_id and transaction_id are not the same